realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -1225,10 +1230,12 @@ private void emitInsertListMethod(JavaWriter writer) throws IOException {
 
         writer.beginControlFlow("while (objects.hasNext())")
                 .emitStatement("object = (%s) objects.next()", qualifiedClassName);
-        writer.beginControlFlow("if(!cache.containsKey(object))");
+        writer.beginControlFlow("if (cache.containsKey(object))")
```

Comment:
I really don't know.  My guess would be that this makes not perceptible difference in execution time.  It almost certainly shaves a few bytes off the size of the executable, though.

I agree that it would be nice to have metrics for time to construct the proxy and time to run common proxy operations.